src/script/standard.cpp

```diff
@@ -90,6 +91,108 @@ static bool MatchMultisig(const CScript& script, unsigned int& required, std::ve
     return (it + 1 == script.end());
 }
 
+// Elements: Matching logic for liquid v1 watchman script
+
+// Copied here from interpreter.cpp to not mess with Core header files more
+bool static CheckMinimalPush(const valtype& data, opcodetype opcode) {
+    // Excludes OP_1NEGATE, OP_1-16 since they are by definition minimal
+    assert(0 <= opcode && opcode <= OP_PUSHDATA4);
+    if (data.size() == 0) {
+        // Should have used OP_0.
+        return opcode == OP_0;
+    } else if (data.size() == 1 && data[0] >= 1 && data[0] <= 16) {
+        // Should have used OP_1 .. OP_16.
+        return false;
+    } else if (data.size() == 1 && data[0] == 0x81) {
+        // Should have used OP_1NEGATE.
+        return false;
+    } else if (data.size() <= 75) {
+        // Must have used a direct push (opcode indicating number of bytes pushed + those bytes).
+        return opcode == data.size();
+    } else if (data.size() <= 255) {
+        // Must have used OP_PUSHDATA.
+        return opcode == OP_PUSHDATA1;
+    } else if (data.size() <= 65535) {
+        // Must have used OP_PUSHDATA2.
+        return opcode == OP_PUSHDATA2;
+    }
+    return true;
+}
+
+static bool MatchLiquidWatchman(const CScript& script)
+{
+    CScript::const_iterator it = script.begin();
+    std::vector<unsigned char> data;
+    opcodetype opcode;
+
+    // Stack depth check for branch choice
+    if (!script.GetOp(it, opcode, data) || opcode != OP_DEPTH) {
+        return false;
+    }
+    // Take in value, then check equality
+    if (!script.GetOp(it, opcode, data) ||
+            !script.GetOp(it, opcode, data) ||
+            opcode != OP_EQUAL) {
+        return false;
+    }
+    // IF EQUAL
+    if (!script.GetOp(it, opcode, data) || opcode != OP_IF) {
+        return false;
+    }
+    // Take in value k, make sure minimally encoded number from 1 to 16
+    if (!script.GetOp(it, opcode, data) ||
+            opcode > OP_16 ||
+            (opcode < OP_1NEGATE && !CheckMinimalPush(data, opcode))) {
+        return false;
+    }
+    opcodetype opcode2 = opcode;
+    std::vector<unsigned char> num = data;
+    // Iterate through multisig stuff until ELSE is hit
```

Comment:
What happens if the stuff iterated through is not just multisig stuff? Same for `Find the ENDIF` down below. Is it safe if someone would produce a script with random other things added in this space?

Reply:
This is only a belt and suspenders call, at various other layers the exact template is checked. It cannot be changed once a network is up and running.